File: iris_start.R
iris <- function(year, 
                 random_seed,
                 initial_larvae,
                 initial_nymphs,
                 initial_adults,
                 initial_infected_larvae,
                 initial_infected_nymphs,
                 activation_rate,
                 iris_main_dir,
                 weather_dir,
                 output_mode,
                 model_name
                 ) {
  
  iris_jar_dir <- paste0(iris_main_dir, "/target/")
  iris_input_dir <- paste0(iris_main_dir, "/input/")
  output_dir <- paste0(iris_main_dir, "/output/", model_name)
  
  if (!dir.exists(output_dir)){
    dir.create(output_dir, showWarnings = FALSE)
  }
  
  iris_exe <- "java.exe"
  iris_jar <- paste0("-jar ", iris_jar_dir, "IRIS-1.0-SNAPSHOT-jar-with-dependencies.jar eu.ecoepi.iris.experiments.AdHocSimulation")
  iris_seed <- paste0("-s ", random_seed)
  iris_adults <- paste0("-a ", initial_adults)
  iris_larvae <- paste0("-l ", initial_larvae)
  iris_nymphs <- paste0("-n ", initial_nymphs)
  iris_infected_larvae <- paste0("-i ", initial_infected_larvae)
  iris_infected_nymphs <- paste0("-j ", initial_infected_nymphs)
  iris_activation_rate <- paste0("-r ", activation_rate)
  iris_weather <- paste0("-w ", weather_dir, "weather_", year, ".csv")
  iris_output <- paste0("-o " , output_dir, "/iris_abundance_", year, ".csv")
  iris_output_mode <- paste0("-m ", output_mode)
  
  iris_run <- paste(iris_exe, 
                    iris_jar, 
                    iris_weather, 
                    iris_seed, 
                    iris_output, 
                    iris_larvae,
                    iris_nymphs,
                    iris_adults,
                    iris_infected_larvae,
                    iris_infected_nymphs,
                    iris_activation_rate,
                    iris_output_mode,
                    sep = " ")
  
  system(iris_run)
}


iris_main_dir <- dirname(rstudioapi::getSourceEditorContext()$path)
location <- "regensburg"
year_start <- 1949
year_end <- 2020
climate_simulation <- FALSE
random_seed <- 42
initial_larvae <- 150
initial_nymphs <- 150
initial_adults <- 150
initial_infected_larvae <- 0
initial_infected_nymphs <- 0
activation_rate <- 0.022
#output_mode <- "csv_timeseries"
#output_mode <- "csv_timeseries_summary"
#output_mode <- "csv_timeseries_summary_habitats"
#output_mode <- "csv_timeseries_nymphs"
#output_mode <- "csv_timeseries_nymphs_habitats"
output_mode <- "csv_timeseries_infection"

weather_dir <- paste0(iris_main_dir, "/input/weather/dwd_", location, "/")

if (climate_simulation) {
  #model_name <- "All_models_daily_mean"
  #model_name <- "CCCma-CanESM2_rcp85_r1i1p1_CLMcom-CCLM4-8-17_v1"
  #model_name <- "CCCma-CanESM2_rcp85_r1i1p1_GERICS-REMO2015_v1"
  #model_name <- "IPSL-IPSL-CM5A-MR_rcp85_r1i1p1_IPSL-WRF381P_v1"
  #model_name <- "IPSL-IPSL-CM5A-MR_rcp85_r1i1p1_KNMI-RACMO22E_v1"
  #model_name <- "IPSL-IPSL-CM5A-MR_rcp85_r1i1p1_SMHI-RCA4_v1"
  #model_name <- "MOHC-HadGEM2-ES_rcp85_r1i1p1_CNRM-ALADIN63_v1"
  #model_name <- "MOHC-HadGEM2-ES_rcp85_r1i1p1_DMI-HIRHAM5_v2"
  #model_name <- "MOHC-HadGEM2-ES_rcp85_r1i1p1_GERICS-REMO2015_v1"
  #model_name <- "MOHC-HadGEM2-ES_rcp85_r1i1p1_ICTP-RegCM4-6_v1"
  #model_name <- "MOHC-HadGEM2-ES_rcp85_r1i1p1_IPSL-WRF381P_v1"
  #model_name <- "MOHC-HadGEM2-ES_rcp85_r1i1p1_KNMI-RACMO22E_v2"
  #model_name <- "MOHC-HadGEM2-ES_rcp85_r1i1p1_MOHC-HadREM3-GA7-05_v1"
  #model_name <- "MOHC-HadGEM2-ES_rcp85_r1i1p1_SMHI-RCA4_v1"
  model_name <- "MPI-M-MPI-ESM-LR_rcp85_r3i1p1_GERICS-REMO2015_v1"
  #model_name <- "MPI-M-MPI-ESM-LR_rcp85_r3i1p1_SMHI-RCA4_v1"
  
  weather_dir <- paste0(iris_main_dir, "/input/climate/", model_name, "/csv_", location, "/")  
} else {
  model_name <- "DWD"
}

for (year in year_start : year_end) {
  iris(year, 
       random_seed,
       initial_larvae,
       initial_nymphs,
       initial_adults,
       initial_infected_larvae,
       initial_infected_nymphs,
       activation_rate,
       iris_main_dir,
       weather_dir,
       output_mode,
       model_name) 
}

File: iris_start.py
import os
import subprocess
import numpy as np

climate_models = [
    "CCCma-CanESM2_rcp85_r1i1p1_CLMcom-CCLM4-8-17_v1",
    "CCCma-CanESM2_rcp85_r1i1p1_GERICS-REMO2015_v1",
    "IPSL-IPSL-CM5A-MR_rcp85_r1i1p1_IPSL-WRF381P_v1",
    "IPSL-IPSL-CM5A-MR_rcp85_r1i1p1_KNMI-RACMO22E_v1",
    "IPSL-IPSL-CM5A-MR_rcp85_r1i1p1_SMHI-RCA4_v1",
    "MOHC-HadGEM2-ES_rcp85_r1i1p1_CNRM-ALADIN63_v1",
    "MOHC-HadGEM2-ES_rcp85_r1i1p1_DMI-HIRHAM5_v2",
    "MOHC-HadGEM2-ES_rcp85_r1i1p1_GERICS-REMO2015_v1",
    "MOHC-HadGEM2-ES_rcp85_r1i1p1_ICTP-RegCM4-6_v1",
    "MOHC-HadGEM2-ES_rcp85_r1i1p1_IPSL-WRF381P_v1",
    "MOHC-HadGEM2-ES_rcp85_r1i1p1_KNMI-RACMO22E_v2",
    "MOHC-HadGEM2-ES_rcp85_r1i1p1_MOHC-HadREM3-GA7-05_v1",
    "MOHC-HadGEM2-ES_rcp85_r1i1p1_SMHI-RCA4_v1",
    "MPI-M-MPI-ESM-LR_rcp85_r3i1p1_GERICS-REMO2015_v1",
    "MPI-M-MPI-ESM-LR_rcp85_r3i1p1_SMHI-RCA4_v1",
]

observers = {
    1: "csv_timeseries",
    2: "csv_timeseries_summary",
    3: "csv_timeseries_summary_habitats",
    4: "csv_timeseries_nymphs",
    5: "csv_timeseries_nymphs_habitats",
    6: "csv_timeseries_infection",
}


def create_output_folder(output_dir):
    os.makedirs(output_dir, exist_ok=True)


def iris(
    year,
    random_seed,
    initial_larvae,
    initial_nymphs,
    initial_adults,
    initial_infected_larvae,
    initial_infected_nymphs,
    initial_rodents,
    initial_infected_rodents,
    activation_rate,
    main_dir,
    jar_dir,
    weather_dir,
    observer,
):

    subprocess.run(
        [
            "java.exe",
            "-jar", jar_dir + "/IRIS-1.0-SNAPSHOT-jar-with-dependencies.jar",
            "eu.ecoepi.iris.experiments.AdHocSimulation",
            "-s", str(random_seed),
            "-a", str(initial_adults),
            "-l", str(initial_larvae),
            "-n", str(initial_nymphs),
            "-i", str(initial_infected_larvae),
            "-j", str(initial_infected_nymphs),
            "-u", str(initial_rodents),
            "-v", str(initial_infected_rodents),
            "-r", str(activation_rate),
            "-w", weather_dir + "weather_" + str(year) + ".csv",
            "-o", output_dir + "/iris_output_" + str(year) + ".csv",
            "-m", observer,
        ],
        check=True,
    )


main_dir = os.path.dirname(os.path.abspath("__file__"))
jar_dir = os.path.abspath(main_dir + "/target")

location = "regensburg"
random_seed = 42
activation_rate = 0.022
observer = observers.get(6)

initial_larvae = 100
initial_nymphs = 100
initial_adults = 150
initial_infected_larvae = 50
initial_infected_nymphs = 50
initial_rodents = 10
initial_infected_rodents = 0

with_dwd = False
with_future_climate = False
with_past_climate = False
with_all_climate_models = True

individual_simulation = True
year_start = 2018
year_end = 2018
with_climate_simulation = False
climate_model = climate_models[13]


if individual_simulation:
    print("individual_simulation:")
    with_dwd = False
    with_past_climate = False
    with_future_climate = False

    years = np.arange(year_start, year_end + 1)

    if with_climate_simulation:
        weather_dir = (
            main_dir + "/input/climate/" + climate_model + "/csv_" + location + "/"
        )
        output_dir = os.path.abspath(main_dir + "/output/" + climate_model)
        create_output_folder(output_dir)
    else:
        weather_dir = main_dir + "/input/weather/dwd_" + location + "/"
        output_dir = os.path.abspath(main_dir + "/output/DWD/")
        create_output_folder(output_dir)

    for year in years:
        iris(
            year,
            random_seed,
            initial_larvae,
            initial_nymphs,
            initial_adults,
            initial_infected_larvae,
            initial_infected_nymphs,
            initial_rodents,
            initial_infected_rodents,
            activation_rate,
            main_dir,
            jar_dir,
            weather_dir,
            observer,
        )


if with_dwd:
    print("with_dwd:")
    years_dwd = np.arange(1949, 2020 + 1)
    weather_dir = main_dir + "/input/weather/dwd_" + location + "/"
    output_dir = os.path.abspath(main_dir + "/output/DWD/")
    create_output_folder(output_dir)

    for year in years_dwd:
        iris(
            year,
            random_seed,
            initial_larvae,
            initial_nymphs,
            initial_adults,
            initial_infected_larvae,
            initial_infected_nymphs,
            initial_rodents,
            initial_infected_rodents,
            activation_rate,
            main_dir,
            jar_dir,
            weather_dir,
            observer,
        )

if with_future_climate:
    print("with_future_climate:")
    years_future_climate = np.arange(2021, 2099 + 1)

    for model in climate_models:
        print(model)
        weather_dir = main_dir + "/input/climate/" + model + "/csv_" + location + "/"
        output_dir = os.path.abspath(main_dir + "/output/" + model)
        create_output_folder(output_dir)

        for year in years_future_climate:
            iris(
                year,
                random_seed,
                initial_larvae,
                initial_nymphs,
                initial_adults,
                initial_infected_larvae,
                initial_infected_nymphs,
                initial_rodents,
                initial_infected_rodents,
                activation_rate,
                main_dir,
                jar_dir,
                weather_dir,
                observer,
            )

if with_past_climate:
    print("with_past_climate:")
    years_past_climate = np.arange(1971, 2020 + 1)

    for model in climate_models:
        print(model)
        weather_dir = main_dir + "/input/climate/" + model + "/csv_" + location + "/"
        output_dir = os.path.abspath(main_dir + "/output/" + model)
        create_output_folder(output_dir)

        for year in years_past_climate:
            iris(
                year,
                random_seed,
                initial_larvae,
                initial_nymphs,
                initial_adults,
                initial_infected_larvae,
                initial_infected_nymphs,
                initial_rodents,
                initial_infected_rodents,
                activation_rate,
                main_dir,
                jar_dir,
                weather_dir,
                observer,
            )

print("Run(s) finished.")
